telemetry_test.go

```diff
@@ -14,332 +14,255 @@ import (
+			srcFiles:    make([]string, 0),
+			data: &execResponseData{
+				SrcLocations: make([]string, 0),
+			},
+		}
+		if err := sfa.initFileMetadata(); err == nil {
+			t.Fatal("this should have thrown an error")
+		}
+		if len(st.logs) != 1 {
+			t.Errorf("there should be 1 telemetry data in log. found: %v", len(st.logs))
+		}
+		if sendErr := st.sendBatch(); sendErr != nil {
+			t.Fatal(sendErr)
+		}
+		if len(st.logs) != 0 {
+			t.Errorf("there should be no telemetry data in log. found: %v", len(st.logs))
+		}
+	})
 }
 
 func TestDisableTelemetry(t *testing.T) {
-	config, err := ParseDSN(dsn)
-	if err != nil {
-		t.Error(err)
-	}
-	config.DisableTelemetry = true
-	sc, err := buildSnowflakeConn(context.Background(), *config)
-	if err != nil {
-		t.Fatal(err)
-	}
-	if err = authenticateWithConfig(sc); err != nil {
-		t.Fatal(err)
-	}
-	if !sc.cfg.DisableTelemetry {
-		t.Errorf("DisableTelemetry should be true. DisableTelemetry: %v", sc.cfg.DisableTelemetry)
-	}
-	if sc.telemetry.enabled {
-		t.Errorf("telemetry should be disabled.")
-	}
+	runSnowflakeConnTest(t, func(sct *SCTest) {
+		if !sct.sc.cfg.DisableTelemetry {
+			t.Errorf("DisableTelemetry should be true. DisableTelemetry: %v", sct.sc.cfg.DisableTelemetry)
+		}
+		if sct.sc.telemetry.enabled {
+			t.Errorf("telemetry should be disabled.")
+		}
+	})
 }
 
 func TestEnableTelemetry(t *testing.T) {
-	config, err := ParseDSN(dsn)
-	if err != nil {
-		t.Error(err)
-	}
-	config.DisableTelemetry = false
-	sc, err := buildSnowflakeConn(context.Background(), *config)
-	if err != nil {
-		t.Fatal(err)
-	}
-	if err = authenticateWithConfig(sc); err != nil {
-		t.Fatal(err)
-	}
-	if sc.cfg.DisableTelemetry {
-		t.Errorf("DisableTelemetry should be false. DisableTelemetry: %v", sc.cfg.DisableTelemetry)
-	}
-	if !sc.telemetry.enabled {
-		t.Errorf("telemetry should be enabled.")
-	}
+	runSnowflakeConnTest(t, func(sct *SCTest) {
+		if sct.sc.cfg.DisableTelemetry {
+			t.Errorf("DisableTelemetry should be false. DisableTelemetry: %v", sct.sc.cfg.DisableTelemetry)
+		}
+		if !sct.sc.telemetry.enabled {
+			t.Errorf("telemetry should be enabled.")
+		}
+	})
 }
 
 func funcPostTelemetryRespFail(_ context.Context, _ *snowflakeRestful, _ *url.URL, _ map[string]string, _ []byte, _ time.Duration, _ bool) (*http.Response, error) {
 	return nil, errors.New("failed to upload metrics to telemetry")
 }
 
 func TestTelemetryError(t *testing.T) {
-	config, err := ParseDSN(dsn)
-	if err != nil {
-		t.Error(err)
-	}
-	sc, err := buildSnowflakeConn(context.Background(), *config)
-	if err != nil {
-		t.Fatal(err)
-	}
-	if err = authenticateWithConfig(sc); err != nil {
-		t.Fatal(err)
-	}
-	sr := &snowflakeRestful{
-		FuncPost:      funcPostTelemetryRespFail,
-		TokenAccessor: getSimpleTokenAccessor(),
-	}
-	st := &snowflakeTelemetry{
-		sr:        sr,
-		mutex:     &sync.Mutex{},
-		enabled:   true,
-		flushSize: defaultFlushSize,
-	}
+	runSnowflakeConnTest(t, func(sct *SCTest) {
+		sr := &snowflakeRestful{
```

Comment:
Inline?

Reply:
Better not. It is easier to read in separation